ops/_tracing/api.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""FIXME docstring."""
+
+from __future__ import annotations
+
+from typing import Any
+
+import ops
+
+from .vendor.charms.certificate_transfer_interface.v1.certificate_transfer import (
+    CertificateTransferRequires,
+)
+from .vendor.charms.tempo_coordinator_k8s.v0.tracing import TracingEndpointRequirer
+
+# NOTE: the shape of certificate_transfer is `{"certificates": list[str]}`
+# it is trivial to read the certificates out manually
```

Comment:
I strongly suspect we will want to replace both of these with our own requirer implementations over time. However, vendoring both to start with seems very reasonable.

Reply:
I've documented the databag shapes and will be pushing this update shortly.
They databags seem pretty simple.

@benhoyt what's your take?

Reply:
I think vendoring them to start with is fine. The main thing is I want to ensure we don't bite off more than we can chew here for a v1 (we've already expanded this project in scope quite a bit - and that's partly me! so I don't want to make it worse).